App/screens/Scan.tsx

```diff
@@ -45,34 +29,50 @@ const Scan: React.FC<ScanProps> = ({ navigation }) => {
         headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
       })
       const message = await res.json()
-      displayPendingMessage()
+
       await agent?.receiveMessage(message)
-    } catch (e: unknown) {
-      throw new Error(t('Scan.UnableToHandleRedirection'))
-    }
-  }
+    } catch (err: unknown) {
+      const error = new BifoldError(
+        'Unable to accept connection',
+        'There was a problem while accepting the connection redirection',
+        1024
+      )
 
-  const handleInvitation = async (url: string): Promise<void> => {
-    displayPendingMessage()
-    const connectionRecord = await agent?.connections.receiveInvitationFromUrl(url, {
-      autoAcceptConnection: true,
-    })
-    if (!connectionRecord?.id) {
-      throw new Error(t('Scan.ConnectionNotFound'))
+      navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
+
+      dispatch({
+        type: DispatchAction.SetError,
+        payload: [{ error }],
+      })
     }
-    setConnectionId(connectionRecord.id)
   }
 
-  useEffect(() => {
-    if (connection?.state === ConnectionState.Complete) {
-      dispatch({
-        type: DispatchAction.ConnectionPending,
-        payload: [{ ConnectionPending: false }],
+  const handleInvitation = async (url: string): Promise<void> => {
+    try {
+      const connectionRecord = await agent?.connections.receiveInvitationFromUrl(url, {
+        autoAcceptConnection: true,
       })
 
+      if (!connectionRecord?.id) {
+        throw new Error(t('Scan.ConnectionNotFound'))
```

Comment:
Should we throw a BifoldError here or generic?

Reply:
Will cleanup.